File: src/lambda.cs


IEnumerable<B> Select<A, B>(Func<A, B> fn, IEnumerable<A> items) =>
    items.Select(fn);

var result = Select(x => x * 100, new List<int>() { 1, 2, 3, 4, 5 });

var printable = string.Join(", ", result.ToList());

Console.WriteLine(printable);

File: src/lambda.py

from typing import TypeVar, Callable, Iterable

A = TypeVar("A")
B = TypeVar("B")

def select(fn: Callable[[A], B], list: Iterable[A]) -> Iterable[B]:
    return map(fn, list)


result: Iterable[int] = select(lambda x: x * 100, [1, 2, 3, 4, 5])

printable = list(result)

print(printable)
